### PR TITLE
Check if file is a symlink with web editor (#3687)

### PR DESCRIPTION
**Pull request by lunny**
* editor checks for symlinks

* translate file_is_a_symlink message

* credit translation author

### CODE REVIEW

**Comment by lunny** [comment]
cherry-pick from gogits/gogs#3687

File: routers/repo/editor.go [hunk 1]
```diff
@@ -186,6 +186,11 @@ func editFilePost(ctx *context.Context, form auth.EditRepoFileForm, isNewFile bo
 				return
 			}
 		} else {
+			if entry.IsLink() {
+				ctx.Data["Err_TreePath"] = true
+				ctx.RenderWithErr(ctx.Tr("repo.editor.file_is_a_symlink", part), EDIT_FILE, &form)
```
Comment (tboerger) [inline routers/repo/editor.go]: You need to replace the constant

**Review by tboerger — CHANGES_REQUESTED** [review]

**Comment by lunny** [comment]
waiting https://github.com/go-gitea/git/pull/20

**Comment by tboerger** [comment]
> waiting go-gitea/git#20

Merged

**Comment by tboerger** [comment]
LGTM, just needs an update of the git module vendor.

**Comment by lunny** [comment]
waiting #450

**Review by tboerger — APPROVED** [review]

**Comment by tboerger** [comment]
> waiting #450

Merged

**Comment by Bwko** [comment]
LGTM

**Pull request merged**